### PR TITLE
use jaspResults as an R package inside Jasp

### PR DESCRIPTION
**Pull request by vandenman**
Related to https://github.com/jasp-stats/INTERNAL-jasp/issues/1314 and the syntax project in general.

@JorisGoosen this turned out to be a lot easier than expected. Am I overseeing something?

This is only a minimal example so I tried to keep the changes as small as possible.

### CODE REVIEW

File: R-Interface/jasprcpp.cpp [hunk 1]
```diff
@@ -71,6 +71,12 @@ static std::string				_R_HOME = "";
 
 bool shouldCrashSoon = false; //Simply here to allow a developer to force a crash
 
+// TODO: this probably won't work on windows
```
Comment (JorisGoosen) [inline R-Interface/jasprcpp.cpp:74]: Haha no, now you've broken all utf-8 in windows ;)

But the next R has UCRT as runtime and thus utf-8 as native encoding so once we switch to that we can strip all of this nonsense anyway.

File: R-Interface/jasprcpp.cpp [hunk 2]
```diff
@@ -114,7 +120,7 @@ void STDCALL jaspRCPP_init(const char* buildYear, const char* version, RBridgeCa
 	runCallbackCB								= callbacks->runCallbackCB;
 	readDataSetCB								= callbacks->readDataSetCB;
 
-
+	// TODO: none of this should pollute the global environment.
```
Comment (JorisGoosen) [inline R-Interface/jasprcpp.cpp:123]: Would it be better to manually make a sub-environment?

File: R-Interface/jasprcpp.cpp [hunk 3]
```diff
@@ -158,27 +164,19 @@ void STDCALL jaspRCPP_init(const char* buildYear, const char* version, RBridgeCa
...
-	rInside["jaspResultsModule"]			= givejaspResultsModule();
-
-	//Adding some functions in R to the RefClass (generator) in the module
-	//To do: move this entirely to zzzWrapper if this wasn't done yet.
-	jaspRCPP_parseEvalQNT("jaspResultsModule$jaspTable$methods(addColumnInfo = function(name=NULL, title=NULL, overtitle=NULL, type=NULL, format=NULL, combine=NULL) { addColumnInfoHelper(name, title, type, format, combine, overtitle) })");
-	jaspRCPP_parseEvalQNT("jaspResultsModule$jaspTable$methods(addFootnote =   function(message='', symbol=NULL, col_names=NULL, row_names=NULL) { addFootnoteHelper(message, symbol, col_names, row_names) })");
+	rInside[".sendToDesktopFunction"]	= Rcpp::XPtr<sendFuncDef>(&sendToDesktopFunction);
+	rInside[".pollMessagesFunction"]	= Rcpp::XPtr<pollMessagesFuncDef>(&pollMessagesFunction);
+	rInside[".baseCitation"]			= baseCitation;
+	jaspRCPP_parseEvalQNT("jaspResults:::setSendFunc(.sendToDesktopFunction)");
```
Comment (JorisGoosen) [inline R-Interface/jasprcpp.cpp:170]: And this actually works?
As in, jaspResults is still sending stuff afterwards?

**Comment by JorisGoosen** [comment]
Looks pretty minimal indeed, nice.

File: R-Interface/jasprcpp.cpp (hunk 2 above)
Comment (vandenman) [reply R-Interface/jasprcpp.cpp:123]: yes, or we assign it into (a subenvironment of) the jaspBase or jaspResults namespace.

File: R-Interface/jasprcpp.cpp (hunk 3 above)
Comment (vandenman) [reply R-Interface/jasprcpp.cpp:170]: I tested descriptives + a few other analyses and those worked.

File: R-Interface/jasprcpp.cpp (hunk 1 above)
Comment (vandenman) [reply R-Interface/jasprcpp.cpp:74]: yeah, I figured, but I didn't want to spend a lot of time on that part before discussing the rest of the approach.

**Pull request closed without merging**